tests/runtime/test_grpc_server.py

```diff
@@ -758,6 +762,20 @@ def test_out_of_range_port(sample_inference_service):
             _assert_connection(grpc.insecure_channel(f"localhost:{free_high_port}"))
 
 
+def test_reflection_enabled(runtime_grpc_server):
+    """This pings the reflection API to ensure we can list the caikit services that are running"""
+    # See https://github.com/grpc/grpc/blob/master/doc/python/server_reflection.md
+    channel = runtime_grpc_server.make_local_channel()
+    reflection_db = ProtoReflectionDescriptorDatabase(channel)
+
+    desc_pool = DescriptorPool(reflection_db)
+    service_desc = desc_pool.FindServiceByName(
+        "caikit.runtime.SampleLib.SampleLibService"
+    )
+    method_desc = service_desc.FindMethodByName("SampleTaskPredict")
+    assert method_desc is not None
+
+
```

Comment:
probably! I wasn't sure the best way to do that though, since we can't generate streaming services yet. That might have to be a lower-level unit test for now that we can remove once we have tests with streaming modules?

Reply:
Fun fact: only the stream_unary case is currently uncovered